runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -5878,8 +5879,19 @@ class INTERPRETER_CLASS
 		case J9VM_CONTINUATION_RETURN_FROM_JIT_MONITOR_ENTER: {
 			rc = tryEnterBlockingMonitor(REGISTER_ARGS, syncObject, J9VM_CONTINUATION_RETURN_FROM_JIT_MONITOR_ENTER);
 			if ((NULL != _currentThread->currentContinuation) && (EXECUTE_BYTECODE == rc)) {
-				void *returnAddress = restoreJITResolveFrame(REGISTER_ARGS);
-				VMStructHasBeenUpdated(REGISTER_ARGS);
+				J9SFJITResolveFrame *resolveFrame = (J9SFJITResolveFrame*)_currentThread->sp;
+				void *returnAddress = resolveFrame->returnAddress;
+				if ((NULL != _currentThread->decompilationStack) && (returnAddress == *(void **)_currentThread->decompilationStack->pcAddress)) {
```

Comment:
This is close to correct.  You should really be checking that pcAddress points to &resolveFrame->returnAddress as the same return address could appear in multiple frames.

Reply:
done